packages/custom-webpack/src/utils.ts

```diff
@@ -1,11 +1,16 @@
-/**
- * check for TS node registration
- * @param file: file name or file directory are allowed
- * @todo tsNodeRegistration: require ts-node if file extension is TypeScript
- */
-export function tsNodeRegister(file: string = '', tsConfig?: string) {
-  if (file && file.endsWith('.ts')) {
-    // Register TS compiler lazily
+const _tsNodeRegister = (() => {
+  let lastTsConfig: string | null | undefined;
+  return (tsConfig?: string) => {
+    // Check if the function was previously called with the same tsconfig
+    if (typeof lastTsConfig !== 'undefined' && (tsConfig ?? null) !== lastTsConfig) {
```

Comment:
I think you can simplify it even more by initializing the `lastTsConfig` to `null`. Thus the only check you'll have to do is the equality check.

Reply:
I wasn't sure if you always call this function with a tsconfig parameter, since it is marked as optional in the calling function. So, to cover all cases, I did it this way. If the tsconfig parameter is always set, this simplification would still cover all cases, but the call signature should be changed.